tools/reckless

```diff
@@ -0,0 +1,684 @@
+        if self.entry is None:
+            for g in unsupported_entry(self.name):
+                for f in tree:
+                    if f['path'] == g:
+                        # FIXME: This should be easier to implement
+                        print(f'entrypoint {g} is not yet supported')
+                        return False
+        dependency_info = ['requirements.txt', 'pyproject.toml']
+        for d in dependency_info:
+            for f in tree:
+                if f['path'] == d:
+                    self.deps = d
+                    break
+            if self.deps is not None:
+                break
+        if not self.entry:
+            return False
+        if not self.deps:
+            return False
+        return True
+
+
+def create_dir(r, directory):
+    """Creation of a directory at path `d` with a maximum new dir depth `r`"""
+    if os.path.exists(directory):
+        return True
+    elif r <= 0:
+        return False
+    elif create_dir(r-1, os.path.split(directory)[0]):
+        os.mkdir(directory, 0o777)
+        print(f'created directory {directory}')
+        if os.path.exists(directory):
+            return True
+
+
+def remove_dir(target):
+    try:
+        shutil.rmtree(target)
+        return True
+    except NotADirectoryError:
+        print(f"Tried to remove directory {target} that does not exist.")
+    except PermissionError:
+        print(f"Permission denied removing dir: {target}")
+    return False
+
+
+class Config():
+    """A generic class for procuring, reading and editing config files"""
+    def obtain_config(self, config_path, default_text, warn=False):
+        """Return a config file from the desired location. Create one with
+        default_text if it cannot be found."""
+        if isinstance(config_path, type(None)):
+            raise Exception("Generic config must be passed a config_path.")
+        assert isinstance(config_path, str)
+        # FIXME: warn if reckless dir exists, but conf not found
+        if os.path.exists(config_path):
+            with open(config_path, 'r+') as f:
+                config_content = f.readlines()
+            return config_content
+        print(f'config file not found: {config_path}')
+        if warn:
+            confirm = input('press [Y] to create one now.\n').upper() == 'Y'
+        else:
+            confirm = True
+        if not confirm:
+            sys.exit(1)
+        parent_path = os.path.split(config_path)[0]
+        # Create up to one parent in the directory tree.
+        if create_dir(1, parent_path):
+            with open(self.conf_fp, 'w') as f:
+                f.write(default_text)
+                # FIXME: Handle write failure
+                return default_text
+        else:
+            print('could not create the parent directory')
+            return None
+
+    def editConfigFile(self, addline, removeline):
+        remove_these_lines = []
+        with open(self.conf_fp, 'r') as reckless_conf:
```

Comment:
Since you just read the file you can also use this:

```
lines = open(self.conf_fp, 'r').readlines()
```

And then use `lines` from thereon, no need for a context manager.